### ISSUE TITLE
NPE when updating comment in issue

### ISSUE THREAD

**Issue by markwoon** [hub4j/github-api#1218]
I'm getting an NPE when trying to update a comment in an issue:

```
java.lang.NullPointerException: null
	at org.kohsuke.github.GHIssueComment.getApiRoute(GHIssueComment.java:149)
	at org.kohsuke.github.GHIssueComment.update(GHIssueComment.java:114)
	...
```

The NPE is because `owner.getRepository()` is returning null.

I can update the issue's body just fine, but cannot update issue comments.

This is with 1.132.

**Comment by bitwiseman** [comment]
@markwoon  What method are you using to get the `GHIssueComment` that is throwing?

**Comment by markwoon** [comment]
Something along the lines of:

```
GitHub github = new GitHubBuilder()
  .withOAuthToken(token)
  .build();
GHRepository repo = github.getRepository(repoName);

GHIssueSearchBuilder builder = github.searchIssues()
  .isOpen()
  .q("foo")
  .q("repo:" + repo.getFullName());
GHIssue issue = builder.list.get(0);

for (GHIssueComment comment : issue.listComments()) {
  if (!comment.getBody().contains("*THIS ISSUE HAS BEEN UPDATED!*")) {
    comment.update(addUpdatedText(comment.getBody()));
  }
}
```

**Comment by bitwiseman** [comment]
Thought it was something like that. 

This doesn't set the `owner` on `GHIssue`:

https://github.com/hub4j/github-api/blob/4247c8a60d809994836d67fe6ce3ad4d5d115406/src/main/java/org/kohsuke/github/GHIssueSearchBuilder.java#L106-L110

There's several things that should be fixed here:
* `GHIssueComment.getApiRoute()` should call `GHIssue.getIssuesApiRoute()` instead of building itself [`GHIssue.getIssuesApiRoute()`](https://github.com/hub4j/github-api/blob/4247c8a60d809994836d67fe6ce3ad4d5d115406/src/main/java/org/kohsuke/github/GHIssue.java#L605-L612) will need to be made package-private for this to work.  
* [`GHIssue.listEvents()`](https://github.com/hub4j/github-api/blob/4247c8a60d809994836d67fe6ce3ad4d5d115406/src/main/java/org/kohsuke/github/GHIssue.java#L754-L758) should be updated to call `GHIssue.getIssuesApiRoute()` instead of referencing `owner`.
* `GHIssue.owner` and `GHIssue.getRepository()` should have an annotation and a comment like [this one](https://github.com/hub4j/github-api/blob/4247c8a60d809994836d67fe6ce3ad4d5d115406/src/main/java/org/kohsuke/github/GHIssue.java#L607) added indicating it maybe null and to use `getIssuesApiRoute()` instead. 
* Perhaps `GHIssue.getRepository()` should check if owner is null and populate it if needed? However, that could result in extra requests.  The alternative is another annotation and a comment like on `owner`, or at least a null check with a message similar to that comment so users know what is going.

**Comment by bitwiseman** [comment]
> * Perhaps `GHIssue.getRepository()` should check if owner is null and populate it if needed? However, that could result in extra requests.  The alternative is another annotation and a comment like on `owner`, or at least a null check with a message similar to that comment so users know what is going.

This is what #1260 implemented. Released in v1.135.   
I think that fixes this issue. @markwoon  Can you check this?

**Comment by markwoon** [comment]
Confirming fixed for my use case in v1.135!
